Generals/Code/GameEngine/Source/GameClient/MessageStream/CommandXlat.cpp

```diff
@@ -1393,7 +1393,8 @@ GameMessage::Type CommandTranslator::evaluateContextCommand( Drawable *draw,
 	// Kris: Now that we can select non-controllable units/structures, don't allow any actions to be performed.
 	const CommandButton *command = TheInGameUI->getGUICommand();
 	if( TheInGameUI->areSelectedObjectsControllable() 
```

Comment:
This function is an enormous mess. We can simplify the fix by writing it like so:

```cpp
...
	const CommandButton *command = TheInGameUI->getGUICommand();

	if (command && command->getCommandType() == GUICOMMANDMODE_PLACE_BEACON)
	{
		msgType = GameMessage::MSG_VALID_GUICOMMAND_HINT;
		TheMessageStream->appendMessage( msgType );
	}
	else if( TheInGameUI->areSelectedObjectsControllable()
...
```